### ISSUE TITLE
Collection change doesn't re-render content in angularjs 1.2.0rc1

### ISSUE THREAD

**Issue by pavelarnost** [Pasvaz/bindonce#12]
Collection change does not re-render content in angularjs 1.2.0rc1. It does work in angularjs 1.0.7 and 1.0.5.

**Comment by spacenick** [comment]
+1

**Comment by Pasvaz** [comment]
Can you test if this issue is still valid and eventually provide more details to reproduce the issue? I've tested both the 1.2.0rc1 and the 1.0.x and they have the same behaviors: when the collection changes the new items are rendered and the old ones, if still present, are not updated. if not present are deleted.

**Comment by pavelarnost** [comment]
Hi,

I found out that it happens with $http service, not with static data. E.g. I have two REST datasources:

/demo?dataset=1
that returns ["one","two"]

/demo?dataset=2
that returns ["1","2","3","4"]

With this code:

```
<script type="text/javascript">
    var IndexCtrl = function($scope, $http) {
        $http.get('/demo?dataset=1').success(function(data) {
            $scope.models = data;
        });

        $scope.changeData = function() {
            $http.get('/demo?dataset=2').success(function(data) {
                $scope.models = data;
            });
        };
    }
</script>

<div ng-controller="IndexCtrl">
<table>
    <tr bindonce="models" ng-repeat="model in models">
        <td bo-text="model"></td>
    </tr>
</table>

<button ng-click="changeData()">change data</button>
</div>

```

after load I see "one two" and after click on button I see "one two 3 4". It looks like that changed data are not updated - with angularjs 1.1.5, I see "1 2 3 4".

**Comment by Pasvaz** [comment]
Thank you for investigating further, I've made this plunker in order to debug your issue but I can't reproduce the bug, can you check it http://plnkr.co/edit/WcDxUq?p=preview

**Comment by pavelarnost** [comment]
Hi,

pls try this plunker:

http://plnkr.co/edit/ToC7Pz?p=preview

I changed only script.js.

**Comment by spacenick** [comment]
Working when you use track by (see a fork of your plunker here : http://plnkr.co/edit/5jX3yq?p=preview) sooo not really sure what's the issue here

**Comment by Pasvaz** [comment]
It works if you initialize 'models' in the current scope: http://plnkr.co/edit/EU3AXv?p=preview I just added `$scope.models = [];`
It seems a scope issue, I'll investigate further and tell you more.

**Comment by spacenick** [comment]
Hey Pasvaz, any update on that?

I'm still facing that issue with 1.2rc2. When my collections changes, the data rendered by bindonce is not updated.. Pretty weird

Thanks man

**Comment by Pasvaz** [comment]
This issue was fixed in Angular 1.2.0, feel free to reopen it if I am wrong.
